### DISCUSSION TITLE
Facing errors with converting Python to APK using the Flet framework

### DISCUSSION THREAD

**Discussion by Maskedviper** [flet-dev/flet#2452] [Q&A]
### Question

We are working on creating an app using Python and the Flet framework, after following several videos and links we are still running into an error in packaging the code as an apk file using the Flet framework

The steps we are following are:
1) Install android studio and set ANDROID_SDK_ROOT
2) Install flutter and set to env AND SET ENV
3) Go to flutter doctor
4) Create the flet app and requirements.txt
5) build to APK

This is the link we are following: 
https://www.youtube.com/watch?v=OqMYfspxKi0&ab_channel=SriEdyNurcahyo

Is there any resources that could be provided or know why we are getting this error?

### Error message

```bash
flet build apk --verbose
usage: flet [-h] [--version] {create,run,pack,publish} ...
flet: error: unrecognized arguments: apk
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment]
Please upgrade Flet to the latest 0.19.0 version.

**Reply by Maskedviper** [reply]
Using the command "pip install flet -upgrade"?

**Reply by FeodorFitsner** [reply]
`pip install flet --upgrade` (double dash)

**Reply by Maskedviper** [reply]
Yes I did that but we are getting the error: "ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied:"

**Reply by FeodorFitsner** [reply]
Would be great to see the full build log.

**Reply by Maskedviper** [reply]
The full build log as in when I try to run the command, "pip install flet --upgrade"

**Reply by wschume** [reply]
Usually you get an access denied error, when you try install stuff into a virtual environment that was created by another user (usually root). Try creating a new virtual environment and install flet there:

```
python -m venv my_new_venv
my_new_venv/bin/pip install flet # linux
my_new_venv\\Scripts\\pip.exe install flet # windows
```

If you are facing issues here, please let me know.

**Comment by Vallekralle** [comment]
I got the same issue here this is my log (in my **requirements.txt** is only flet and no more, I already tried it with everything, so with the command `pip freeze > requirements.txt` both times it shows the same error):
```
Packaging Python app...Building package executable... (1.8s)
Built serious_python:main.
Running package command
Creating asset directory: C:\Users\kunzv\AppData\Local\Temp\flet_flutter_build_dOphibiyoc\app
Copying Python app from C:\Users\kunzv\OneDrive\Dokumente\YouTubeMobileDownloader\YouTubeDownloader\Downloader to C:\Users\kunzv\AppData\Local\Temp\serious_python_temp1f82b35f
Error: FileSystemException: Failed to decode data using encoding 'utf-8', path = 'C:\Users\kunzv\AppData\Local\Temp\serious_python_temp1f82b35f\requirements.txt'
Deleting temp directory C:\Users\kunzv\AppData\Local\Temp\serious_python_temp1f82b35f
Flet app package app/app.zip was not created.
Deleting Flutter bootstrap directory C:\Users\kunzv\AppData\Local\Temp\flet_flutter_build_dOphibiyoc
Error building Flet app - see the log of failed command above.
```

_I also took a screenshot if you want to see it in another format..._

![Flet apk build error](https://github.com/flet-dev/flet/assets/108894835/7e9ead3a-b4e1-4ff1-839f-e4e6cc0e4cc2)

**Reply by FeodorFitsner** [reply]
Do not use pip freeze: https://flet.dev/docs/guides/python/packaging-app-for-distribution#project-structure

<img width="981" alt="image" src="https://github.com/flet-dev/flet/assets/5041459/443850c4-3050-408f-933c-811bc255a6c8">

**Reply by Vallekralle** [reply]
You are not a programer you are a brother, thank you very much... I should have read the docs first 😬

**Comment by New19User** [comment]
> Do not use pip freeze: https://flet.dev/docs/guides/python/packaging-app-for-distribution#project-structure
> <img alt="image" width="981" src="https://private-user-images.githubusercontent.com/5041459/300552411-443850c4-3050-408f-933c-811bc255a6c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY1Njc0MjYsIm5iZiI6MTcwNjU2NzEyNiwicGF0aCI6Ii81MDQxNDU5LzMwMDU1MjQxMS00NDM4NTBjNC0zMDUwLTQwOGYtOTMzYy04MTFiYzI1NWE2YzgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjlUMjIyNTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODA1YTM4OTI5OTAxZmY0ZWY4MjEzNmY1MGExMmQzYjNhNDgwMzJlNGE5NzYwZjcyN2M2ODUwNzg0NjY4NjE4ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.3g9uoRhXxEaQ_F2vDN_XKGtsLsmrG2LWCJDZPj2nbUc">

I hand picked my requirements, still yet for more than 2 weeks i have been building with flet, i haven't been able to build an apk, this is always the result, i have tried using the vs command line, windows command line as an admin, still always the same error. 
![build failed](https://github.com/flet-dev/flet/assets/130806630/a6e6a805-2aed-499f-951f-2406454d6dec)
